packages/google_maps_flutter/google_maps_flutter_web/README.md

```diff
@@ -72,3 +74,27 @@ Indoor and building layers are still not available on the web. Traffic is.
 Only Android supports "[Lite Mode](https://developers.google.com/maps/documentation/android-sdk/lite)", so the `liteModeEnabled` constructor argument can't be set to `true` on web apps.
 
 Google Maps for web uses `HtmlElementView` to render maps. When a `GoogleMap` is stacked below other widgets, [`package:pointer_interceptor`](https://www.pub.dev/packages/pointer_interceptor) must be used to capture mouse events on the Flutter overlays. See issue [#73830](https://github.com/flutter/flutter/issues/73830).
+
+### Custom marker icons
+
+Images are not automatically scaled according to the `imageAspectRatio` value of the `BitmapDescriptor` in web, so the `size` parameter must be used to scale the marker when using the `AssetMapBitmap` or `BytesMapBitmap` methods.
```

Comment:
The biggest issue (to this day) of custom markers on the web is that the maps SDK expects an image rendered @ "physical resolution", but then it needs help scaling it down to "logical resolution", so they look sharp (but not giant).

Is there any way of expressing that with the new primitives? Can users create a `BytesMapBitmap` image at 2x or 3x, then pass it with a size to the web? Do we still need this? Is there a good example for the web?

Is this behavior consistent with other platforms, or will people need some special code to render images that look sharp across all platforms?

Reply:
Scaling has some challenges on the web, and this PR currently supports scaling only through the size parameter, which is specified in logical pixels.

With AdvancedMarkers, it should be possible to specify an image element that has width parameter as a percentage, allowing for the scaling of the image based on the given pixel ratio. However, the Google Maps binding library does not yet include bindings for advanced markers: [Google Maps package on pub.dev](https://pub.dev/packages/google_maps) and [Advanced Markers documentation](https://developers.google.com/maps/documentation/javascript/advanced-markers/overview).

Scaling markers using percentages—which could be calculated from the pixel ratio and dpi—is not currently possible because of following problems:

Marker size is specified using two parameters:

```dart
  icon.size = <gmaps.Size object>;
  icon.scaledSize = <gmaps.Size object>;
```
where icon.size is applied to the div wrapper, and icon.scaledSize is applied to the image.

```dart
  external Size(
    num? width,
    num? height, [
    String? widthUnit, // ignore: unused_element
    String? heightUnit, // ignore: unused_element
  ]);
```
widthUnits can be used to give type of the unit (like `px`, `%`, `vh`, `vw`)

By default, Google Maps uses the image size for the div size, which is overridden if icon.size or icon.scaledSize is specified. It's not possible to set scaledSize without also overriding size. Giving size without knowing the right size of the image is not possible.

Earlier, I had a version where the image size was read from the image header data after loading, and it was automatically scaled using image scale and given pixel ratio. However, this method is not ideal by default, as it is not a very performance-efficient option, as the size need to be read from the image header. Note that with advanced markers it should be possible to to add onLoad event handler for image object given for the map, and tune the image element settings after the image is loaded.

So, with current version of google maps binding library, I don't see other options than support the size parameter on web. Of course in this case developers need to know the size of the asset; and for web assets this can be tricky without affecting the performance.

Reply:
> So if people know the image size, they can give it in proper size.
for example if the website has pixel ratio is 2.0 users just need to give 100x100 (physical pixel) image and add it as marker with size 50x50 (logical pixels).

As long as people can create their own marker as 100x100 (physical) and then tell google maps that it needs to be 50x50 (logical), I'm totally fine. I didn't see the separation of the `size` parameter on the new image classes, but they probably require a different process to draw the pixels, which then are passed to the map?

Reply:
Actually,
The previous method of utilizing the `image` package was not good because it used Dart code to access the image's dimensions (width and height) as discussed [here](https://github.com/flutter/plugins/pull/6805#discussion_r1095855056). Meaning the image was loaded and decoding the image in both Dart and the browser, which wasn't inefficient. 

Alternatively, by leveraging the [HTMLImageElement](https://pub.dev/documentation/web/latest/web/HTMLImageElement-extension-type.html) to load the image and extracting its dimensions on its onLoad callback function, performance should not be better. I'll test and betchmark if this approach works well and do not affect the overall performance of marker modification with the plugin. This approach must take care of the addition/modification of multiple markers to the map in a batch, avoiding the need to wait for each image to load sequentially.